Realm/RLMRealm.mm

```diff
@@ -448,9 +448,19 @@ + (instancetype)realmWithConfiguration:(RLMRealmConfiguration *)configuration
     }
 
     if (configuration.seedFilePath && ![RLMRealm fileExistsForConfiguration:configuration]) {
-        [[NSFileManager defaultManager] copyItemAtURL:configuration.seedFilePath
-                                                toURL:[configuration fileURL]
-                                                error:error];
+        bool didCopySeed = false;
+        bool didLock = DB::call_with_lock(configuration.seedFilePath.path.UTF8String,
+                                          [&configuration, &error, &didCopySeed](auto const&) {
+            didCopySeed = [[NSFileManager defaultManager] copyItemAtURL:configuration.seedFilePath
+                                                                  toURL:configuration.fileURL
+                                                                  error:error];
+        });
+        if (!didLock) {
+            @throw RLMException(@"Cannot copy seed Realm at path '%@' while it is already open.", configuration.seedFilePath);
```

Comment:
This isn't an error. If call_with_lock() returns false, it just means that someone else created the file between when we checked if it existed and when we tried to get the lock. It's fine if that happens, and we should just open the file.